app/test_api.py

```diff
@@ -0,0 +1,178 @@
+        print(response.data)
+        self.assertEqual(response.status_code, 400)
+    
+    def test_signup_taken_username(self):
+        UserInfo = json.dumps({
+            "username": "sampleusername",
+            "email": "sampleemail",
+            "password": "Yunis",
+
+            "username": "sampleusername",
+            "email": "sampleemail",
+            "password": "Yunis",
+        
+        })
+        response = self.client().post('/api/v1/users', data=UserInfo,
+                                      content_type='application/json')
+        print(response.data)
+        self.assertEqual(response.status_code, 400)
+    
+    def test_signup_no_password(self):
+        UserInfo = json.dumps({
+            "username": "sampleusername",
+            "email": "sampleemail",
+            "password": "",
+            
+        })
+        response = self.client().post('/api/v1/users', data=UserInfo,
+                                      content_type='application/json')
+        print(response.data)
+        self.assertEqual(response.status_code, 400)
+    
+    def test_signup_caps_password(self):
+        UserInfo = json.dumps({
+            "username": "sampleusername",
+            "email": "sampleemail",
+            "password": "yuns8888#",
+            
+        })
+        response = self.client().post('/api/v1/users', data=UserInfo,
+                                      content_type='application/json')
+        print(response.data)
+        self.assertEqual(response.status_code, 400)
+    
+    def test_signup_number_password(self):
+        UserInfo = json.dumps({
+            "username": "sampleusername",
+            "email": "sampleemail",
+            "password": "Yunisabcde",
+            
+        })
+        response = self.client().post('/api/v1/users', data=UserInfo,
+                                      content_type='application/json')
+        print(response.data)
+        self.assertEqual(response.status_code, 400)
+    def test_signup_special_password(self):
+        UserInfo = json.dumps({
+            "username": "sampleusername",
+            "email": "sampleemail",
+            "password": "Yunisab2",
+            
+        })
+        response = self.client().post('/api/v1/users', data=UserInfo,
+                                      content_type='application/json')
+        print(response.data)
+        self.assertEqual(response.status_code, 400)
+    
+    def test_signup_length_password(self):
+        UserInfo = json.dumps({
+            "username": "sampleusername",
+            "email": "sampleemail",
+            "password": "Yunis",
+            
+        })
+        response = self.client().post('/api/v1/users', data=UserInfo,
+                                      content_type='application/json')
+        print(response.data)
+        self.assertEqual(response.status_code, 400)
+    
+    def test_signup_string_password(self):
+        UserInfo = json.dumps({
```

Comment:
Observe python naming conventions, throughout
```suggestion
        user_Info = json.dumps({
```

Reply:
Fixed